src/core/middlewares/middlewares.go

```diff
@@ -78,5 +79,6 @@ func MiddleWares() []beego.MiddleWare {
 		artifactinfo.Middleware(),
 		security.Middleware(),
 		readonly.Middleware(readonlySkippers...),
+		metric.Middleware(),
```

Comment:
Move this middleware up so it can take the time spent on other middlewares into account, but it should run after request ID.